### PR TITLE
fix(merge-default): use any non-null object type instead of Record

### PR DESCRIPTION
**Pull request by kyranet**

### CODE REVIEW

**Review by cfanoulis — APPROVED** [review]
LGTM

**Review by favna — APPROVED** [review]

File: src/lib/mergeDefault.ts [hunk 1]
```diff
@@ -3,21 +3,21 @@ import { isObject } from './isObject';
...
 /**
  * Sets default properties on an object that aren't already specified.
  * @since 0.5.0
  * @param def Default properties
  * @param given Object to assign defaults to
  */
-export function mergeDefault<A extends KeyedObject, B extends Partial<A>>(defaults: A, given?: B): DeepRequired<A & B> {
+export function mergeDefault<A extends NonNullObject, B extends Partial<A>>(defaults: A, given?: B): DeepRequired<A & B> {
 	if (!given) return deepClone(defaults) as DeepRequired<A & B>;
 	for (const key in defaults) {
```
Comment (vladfrangu) [inline src/lib/mergeDefault.ts]: While you're at it, why are we using for-in?

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: src/lib/mergeDefault.ts (hunk 1 above)
Comment (kyranet) [reply src/lib/mergeDefault.ts]: Resolved in 101b66d.

Comment (vladfrangu) [reply src/lib/mergeDefault.ts]: Come again? you're still using `in`

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**